src/main/telemetry/smartport.c

```diff
@@ -578,7 +369,7 @@ void handleSmartPortTelemetry(void)
         }
 
         if (smartPortMspReplyPending) {
-            smartPortMspReplyPending = smartPortSendMspReply();
+            smartPortMspReplyPending = sendMspReply(SMARTPORT_PAYLOAD_SIZE, &smartPortSendMspResponse);
```

Comment:
I suspect the state issue is here

Reply:
Well possible, I can test some more on the weekend. Now it's nightie nights for me.